components/script_plugins/unrooted_must_root.rs

```diff
@@ -206,7 +206,7 @@ impl<'a, 'b, 'tcx> visit::Visitor<'tcx> for FnDefVisitor<'a, 'b, 'tcx> {
     fn visit_pat(&mut self, pat: &'tcx hir::Pat) {
         let cx = self.cx;
 
-        if let hir::PatKind::Binding(hir::BindingMode::BindByValue(_), _, _, _) = pat.node {
+        if let hir::PatKind::Binding(hir::BindingAnnotation::Unannotated, _, _, _) = pat.node {
```

Comment:
This should also check `BindingAnnotation::Mutable` which is used for `mut x` bindings.

This is also susceptible to false positives once "default binding modes" is implemented (rust-lang/rust#42640); we should probably add a fixme comment and/or file a bug about that.